subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractConsoleFunctionalSpec.groovy

```diff
@@ -46,6 +46,24 @@ class AbstractConsoleFunctionalSpec extends AbstractIntegrationSpec {
         return styledString
     }
 
+    /**
+     * Grabs the grouped output of a task
+     * @param taskName a fully qualified task name e.g. :project:task1
+     * @return
+     */
+    protected String taskOutputFrom(String taskName) {
```

Comment:
I think we'll eventually get to a model that represents the rich output you described, but it will be backed by fancy regexes.  Unless we mock the output itself, which to some degree invalidates the 'functional' part of the spec.  For now, I think I'll take up your suggestion of moving the logic into its own class, but not yet embedding it into the `OutputScrapingExecutionResult`.  As the API develops and settles down, we can add delegate methods as we see fit.